src/Google.Bigquery.V2/BigqueryDataset.cs

```diff
@@ -22,97 +22,115 @@ namespace Google.Bigquery.V2
     /// A dataset within Bigquery.
     /// </summary>
     /// <remarks>
-    /// This class wraps the underlying HTTP API resource and retains a reference to the original
+    /// This class wraps the underlying REST API resource and retains a reference to the original
     /// client, allowing for simpler dataset-oriented operations.
     /// </remarks>
     public sealed class BigqueryDataset
```

Comment:
Not sure. We could just make the constructor public, and let users pass in their faked client. I think that's simpler than making *everything* into an abstract class. (Where would it end?)